### DISCUSSION TITLE
Example configs for custom voice training? + few questions

### DISCUSSION THREAD

**Discussion by CZ-DannyK** [coqui-ai/TTS#1298] [General Q&A]
Hello guys,

i have recently found this interesting project. I would like to use this to train custom voice, but i am not exactly sure, how to properly do this.

I have managed to get this project running on Windows with newer Cuda and PyTorch (that support my GK), update trainer.py to use spawn instead of fork for multiprocessing and in overall it seems to be working. But i got stuck a bit on how to properly train NN to my custom voice? I have several questions: 

1, Do i need to train both TTS and Vocoder networks to my audio samples?
2, What exactly are speakers?
3, I have several thousands of audio samples in 44100 sample rate of various length (short, long sentences), generated metadata.csv for ljspeech formatter [sample_name, text, cleaned_text]. - Could someone help me with configs? Or provide some example config i can use for test training (either TTS or Vocoder)?
4, How many samples to use (i am aware of it can differ based on used models and other factors, but some general rule of thumb?)
5, What is optimal avg loss when training? Or what values to watch?
6, I will be glad for any other advices or help to get started.

Thanks for help
D.

Update:
- I will answer some points myself (if i am wrong, feel free to correct me)

1, Yes and no - In overall yes, both TTS and Vocoder models should be trained on same dataset to get best results. It might happen some Vocoder will work fine by default for custom dataset, but having trained one should always work better.
Small explanation of what is what:

TTS - Converts Text to Spectrum (2D graph)
Vocoder - Converts Spectrum to Final Audio

Together they make optimal conversion chain.

2, As far as i understood via various sources, Speakers comes in play with mutlispeaker datasets. Each line can basically have speaker name, and one model will be trainer for multiple voices (speakers). 

3, I started training of TTS with this recipes: [https://github.com/coqui-ai/TTS-recipes/tree/master/LJSpeech/DoubleDecoderConsistency]( https://github.com/coqui-ai/TTS-recipes/tree/master/LJSpeech/DoubleDecoderConsistency)
I am around 76k steps, audiosamples (EvalAudio, TrainAudios) in tensorboard are getting to be understandable (i hear words and recognize voice). Generated ones (TestAudio-x) are slowly getting shape. They have artefacts, sounds robotic, but it still not enough steps i guess and its using default Griffin-Lim vocoder.
I will try to train it around 200k steps, see results and then start training Vocoder model.
P.S. I have resampled original 44100 wavs to 22050 as most used sample rate (i used resample.py in TTS project), then computed statistics (used compute_statistics.py in TTS project)

4, I have used for now 12k+ audio samples. Should be enough i hope. But they would deserve probably some cleaning as they have often pauses between parts of sentence, but will see after training and then maybe i will start adjusting dataset for fine tuning.

5, Still no clue. I am watching avg_loss value, if its still going down, i consider it  good sign.

**Comment by erogol** [comment]
have you checked https://tts.readthedocs.io/en/latest/

**Reply by CZ-DannyK** [reply]
Yes, i did, except for papers i went through all sections, but still could not find answers for my questions.

**Reply by ProgrammingLife** [reply]
I also need some little example that would show how to train model using actual TTS version. Including all working config and dataset files inplace. It's hard to understand the order of commands and the settings for beginners.

**Reply by CZ-DannyK** [reply]
Yeah, its quite hard to begin (probably this is more advanced project and some ML knowledge is expected), but still more informations would be fine. I will edit first post and add some knowledge i gathered during last few days so it can help someone else as well.
